src/conn/conn_prefetch.c

```diff
@@ -88,22 +84,23 @@ __wt_prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
 
     while (!TAILQ_EMPTY(&conn->pfqh)) {
         __wt_spin_lock(session, &conn->prefetch_lock);
-        locked = true;
         pe = TAILQ_FIRST(&conn->pfqh);
 
         /* If there is no work for the thread to do - return back to the thread pool */
-        if (pe == NULL)
+        if (pe == NULL) {
+            __wt_spin_unlock(session, &conn->prefetch_lock);
             break;
+        }
 
         TAILQ_REMOVE(&conn->pfqh, pe, q);
         --conn->prefetch_queue_count;
 
```

Comment:
Add comment:
"We increment this in the lock as the queue draining thread expects that behaviour."